source/_integrations/enphase_envoy.markdown

```diff
@@ -135,33 +184,121 @@ This integration provides several values suitable for the energy dashboard:
+action: switch.turn_on
+data: {}
+target:
+  entity_id:
+    - switch.no2
+    - switch.nc2
+```
+
+### Action `number.set_value`
+
+This action changes the setting for the Enpower `Reserve battery level`.
+
+| Data attribute | Optional | Description |
+| - | - | - |
+| `entity_id` | no | Name(s) of entities, e.g., `number.enpower_12345678901001_reserve_battery_level`. |
+| `value` | no | The target Value between 0 and 100 to set Enpower reserve battery level to. |
+
+Example:
+
+```yaml
+action: number.set_value
+target:
+  entity_id: number.enpower_12345678901001_reserve_battery_level
+data:
+  value: "25"
+```
+
+### Action `select.select`
+
+This action changes the:
+
+- Relays relay_mode, grid_action, micro_grid_action or generator_action
+- Battery storage mode
+
+| Data attribute | Optional | Description |
+| - | - | - |
+| `entity_id` | no | Name(s) of entities, e.g., `select.nc2_generator_action`. |
+| `option` | no | For relay modes: `powered`, `not_powered`, `schedule`, `none`. <br> For storage modes: `backup`, `self_consumption`, `savings` |
+
+Example:
 
-## Debug logs and diagnostics
+```yaml
+action: select.select_option
+target:
+  entity_id:
+    - select.nc2_generator_action
+data:
+  option: not_powered
 
-This integration provides debug logs and diagnostics reports as described in the [Home Assistant troubleshooting pages](/docs/configuration/troubleshooting/#debug-logs-and-diagnostics).
+action: select.select_option
+target:
+  entity_id:
+    - select.enpower_12345678901001_storage_mode
+data:
+  option: backup
+```
+
+{% note %}
+Technically `select.first`, `select.last`, `select.previous`, `select.next` are available as well, but as there's no logical sequence in the values to select, their use is not advocated.
+{% endnote %}
+
+## Know issues and limitations
+
+- [Envoy firmware](https://enphase.com/installers/resources/documentation/communication?&f[0]=document%3A217) versions come with changing behavior, features and issues. Firmware is pushed to the Envoy by Enphase, while 'not always communicated in detail upfront'. This may result in sudden changes in the Envoy behavior and is always accompanied by an outage while Envoy is being updated.
+- As of Envoy firmware  8.2.4225 the Envoy does no longer support setting battery modes, Enable/Disable charge from grid or changing reserve battery level through the local REST API used by HA. Until a resolution is found, you will need to use the Enphase APP to control these.
+
+## Troubleshooting
+
+- The Envoy should not be both on your local LAN and local Wifi at the same time. This may cause auto discovery to switch back and forth between the interfaces resulting in brief outages every 30 minutes. If you experience these frequent brief outages, make sure only 1 interface is used.
```

Comment:
```suggestion
- The Envoy should not be both on your local LAN and local Wi-Fi at the same time. This may cause auto-discovery to switch back and forth between the interfaces resulting in brief outages every 30 minutes. If you experience these frequent brief outages, make sure only 1 interface is used.
```